### ISSUE TITLE
Kubernetes cluster pods as Jenkins Dynamic Build Agents

### ISSUE THREAD

**Issue by mohanss08** [GoogleCloudPlatform/continuous-deployment-on-kubernetes#169]
**Environment: (On Premise):-**
```
Jenkins Master: Standalone Physical server.
Kubernetes Jenkins plugin - 1.23.3
K8S-Master Server – CentOS-7.8- Standalone Physical server & K8S - (v1.18.6)
K8S-Worker Server – Ubuntu-16.04 LTS Standalone Physical server.
Docker Private registry – Nexus-Sonatype
```
Below is my Kubernetes Jenkins plugin configuration.

**Jenkins Plugin Configuration**
Pod Label
Key: jenkins
Value: slave

**Pod Templates**
Name: kubu
Labels: kubupods
Usage: Use this node as much as possible

**Container Template**
Name: jenkins-slave
Docker image: nexus-server:4546/ubuntu-16:version-1
Working Directory: /home/jenkins

**ImagePullSecrets**
Name: regcred

**Below is my Jenkins scripted pipeline Job’s code.**

```
node ("K8S-Worker-Node-1") {  
        stage('Code Checkout') {                
        echo 'PipeLine Code Checkout....'
	checkout([$class: 'GitSCM', branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', depth: 0, noTags: false, reference: '', shallow: false, timeout: 120]], submoduleCfg: [], userRemoteConfigs: [[url: 'ssh://gerritadmin@gerrit-server:29418/test/project/repository']]])
        }
    }	

node ("kubupods") { 
    	stage('Build') { 
            echo 'PipeLine Build on K8S Pod...'	
            sh 'cat /etc/issue'
			sh label: '', script: '''cd $WORKSPACE/src/
            gradle clean build'''
        }
    		
        stage('Static Code Analysis') {    
            echo 'SonarQube Analysis...'
            sh 'cat /etc/issue'
            sh 'hostname'
				
   	        }
        }
```

Basically first I need to download the source to my `K8S-Worker-Node-1` system and further stages (`Build & Static Code Analysis`) need to be performed on `Kubernetes Pods`.  Here dynamically pods need to use our private image and mount the Jenkins workspace from `K8S-Worker-Node-1 system `and `perform the stages`. I have no success with this approach, Can someone help me with this? It would be helpful for our continues integration with Kubernetes.

**Comment by mohanss08** [comment]
This link has examples and helped me to setup - https://github.com/jenkinsci/kubernetes-plugin
